node/node.go

```diff
@@ -978,6 +984,45 @@ func (n *Node) ProcessBatch(
 	return signature, nil
 }
 
+// Shutdown gracefully shuts down the node and its components
+func (n *Node) Shutdown() {
```

Comment:
This method should also shut down LittDB (`db.Close()`). The LittDB instances is embedded inside the `ValidatorStore`.

Although not strictly necessary from a correctness point of view, shutting down the DB allows it to fully flush data to disk, which means it doesn't need to do recovery operations at the next startup. A hard shutdown can result in partially corrupted files (recoverable), and could conceivably take nontrivial time for LittDB to parse and fix the problem.

When doing this, important to wait for all threads that might be writing to the DB to be stopped. If you stop the DB then write to it, the DB will clutter the log with nasty messages.